src/cleaning_functions.py

```diff
@@ -24,26 +24,29 @@ def get_label_encoded_feature(feature):
     return [label_encoded_feature]
 
 
-def get_one_hot_encoded_feature(feature):
+def get_one_hot_encoded_feature(feature, feature_name):
```

Comment:
knowing the feature name is of course essential, but i was referring to them being two separate variables, where they should be encapsulated in on thing

having them separat makes it awkward do work with them - i.e. (exaggerating) i just want to get the company feature, now i have to (1) go through the list of column names, (2) find all company substrings, (3) remember their indicies, and then go to the separate data variable and (4) get the respective indicies

where it would be much cleaner to encapsulate that in an object just say `myfeatures.company` which returns the values of col_company_1, col_company_2...

the context is in the attribute, which is serialized with the object, we might as well just pickle it - don't have to store that separately as a string. 

as for the specific functions, they don't have to care about the feature names:
```
companies = get_encoded(raw_companies)
job_titles = get_encoded(raw_job_titles)
vs.
rnd_name_1 = get_encoded(raw_companies,'companies')
rnd_name_2 = get_encoded(raw_job_titles, 'job_titles')
```
you know its the companies because you have the variable reference, which isnt lost when you serialize the object properly

the super clean version would be to remove the data from the base model entirely and have one
data object, which you pass to a model (this would also speed up everything and use WAY less memory) (i'm also seeing that just now - hindsight is 20/20) this object can easily be serialized and has a clean interface for accessing data (also the train/test split)